### ISSUE TITLE
ItemClickListener add support for getting column from event

### ISSUE THREAD

**Issue by petrixh** [vaadin/vaadin-grid-flow#725]
Sometimes it is handy to be able to determine which column on a grid row was clicked for instance: 

- for example a clickable link in the cell and you want to handle it on the server-side, even if you click next to the link. 
- Or you have a checkbox in a cell and want to toggle it, even if you do not directly click it

This was possible in the Vaadin 8 Grid itemClickEvent but is missing in the v14 itemClickEvent

Tested: 
Vaadin 14.0.0.rc6

**Comment by JohannesLaidler** [comment]
The ItemClickEvent should also give access to the EditorComponent in order to focus the correct component, when the user clicks on a cell. 

There is currently a workaround for accessing the Column ([Forum-Post](https://vaadin.com/forum/thread/17585345/get-the-grid-column-by-clicklistener)). But `getEditorComponent` **returns null**, if the `EditorComponent ` is set with 
`setEditorComponent(SerilizableFunction<T,? extends Component> function)` instead of `setEditorComponent(Component component)` .
The Java Doc for `getEditorComponent` currently says this is intended. That's why I wonder how else I should get access to the `EditorComponent` . I wrote a workaround with `AttachListeners` on my `EditorComponents` but it's rather slow. 

```
Gets the editor component that is used for this column.
Returns:
    the editor component, or null if no component isset, or if it was set by using setEditorComponent(SerializableFunction).
```

I'm using vaadin 14.0.5

**Comment by TatuLund** [comment]
Is this duplicate of https://github.com/vaadin/vaadin-grid-flow/issues/612 ?

**Comment by pekam** [comment]
This feature will be included in `vaadin-grid-flow 4.1.0`. That version is targeted for Vaadin 14.2 (not the next minor) in order to ensure a proper pre-release/validation period for this and the other new Grid features.
